pkgs/shells/zsh/zsh-powerlevel10k/default.nix

```diff
@@ -0,0 +1,39 @@
+{ stdenv, fetchFromGitHub, substituteAll, pkgs }:
+
+# To make use of this derivation, use
+# `programs.zsh.promptInit = "source ${pkgs.zsh-powerlevel10k}/share/zsh-powerlevel10k/powerlevel10k.zsh-theme";`
```

Comment:
`nixos/modules/programs/zsh/oh-my-zsh.nix` sets `programs.zsh.interactiveShellInit` instead of `programs.zsh.promptInit` - I wonder what's the difference?

Also, what about adding a small NixOS module?